app/streams/publishing_api/message_handler.rb

```diff
@@ -7,39 +7,61 @@ def self.process(*args)
 
   def initialize(message)
     @message = message
-
-    @new_item = PublishingAPI::MessageAdapter.to_dimension_item(message)
-    @old_item = Dimensions::Item.find_by(base_path: new_item.base_path, latest: true)
   end
 
   def process
-    return unless new_item.older_than?(old_item)
-
-    if is_links_update?
-      grow_dimension! if links_have_changed?
-    else
-      grow_dimension!
+    items.each do |item|
+      if is_links_update?
+        item.process_links!
+      else
+        item.process!
+      end
     end
   end
 
 private
 
-  attr_reader :message, :new_item, :old_item
+  attr_reader :message
 
-  def is_links_update?
-    routing_key = message.delivery_info.routing_key
-    routing_key.ends_with?('.links')
+  def get_old_items(new_base_paths)
+    content_id = message.payload["content_id"]
+
+    # Get everything with the same content id, so we can deprecate parts
+    # that no longer exist, and get everything with the same base path
+    # to maintain the constraint of only one latest item at each base path
+    Dimensions::Item.where(content_id: content_id, latest: true).or(
+      Dimensions::Item.where(base_path: new_base_paths, latest: true)
+    )
   end
 
-  def links_have_changed?
-    current_links = old_item.raw_json['expanded_links'].deep_sort
-    new_links = new_item.raw_json['expanded_links'].deep_sort
+  def items
+    adapter = PublishingAPI::MessageAdapter.new(message)
+    subpages = adapter.subpages_by_base_path
```

Comment:
It is not clear to me why we have this logic. 
Wouldn't it be enough with:

1. Get all the items for the message (using the adapter)
2. For each new item:

```ruby
PublishingAPI::ItemHandler.new(
  old_item: old_item #Find the old_item for the `new_item.base_path`,
  new_item: new_item,
#  subpage: subpage Not needed, as it would be the `new_item.base_path`
)
```

I know there have to be reasons for the way you have written this code... I would love to find better ways to make it more obvious for the readers.